### PR TITLE
Add decoding for diva delegation

### PR DESCRIPTION
**Pull request by yabirgb**
image

### CODE REVIEW

**Comment by LefterisJP** [comment]
> image

WTF? Why not me? I am there!

**Comment by yabirgb** [comment]
Diff on the globaldb

```
INSERT INTO assets(rowid,identifier,name,type) VALUES(5218,'eip155:1/erc20:0xBFAbdE619ed5C4311811cF422562709710DB587d','Diva Token','C');
INSERT INTO common_asset_details(rowid,identifier,symbol,coingecko,cryptocompare,forked,started,swapped_for) VALUES(5218,'eip155:1/erc20:0xBFAbdE619ed5C4311811cF422562709710DB587d','DIVA','','',NULL,1687892411,NULL);
INSERT INTO evm_tokens(rowid,identifier,token_kind,chain,address,decimals,protocol) VALUES(4324,'eip155:1/erc20:0xBFAbdE619ed5C4311811cF422562709710DB587d','A',1,'0xBFAbdE619ed5C4311811cF422562709710DB587d',18,NULL);
```

File: docs/changelog.rst [hunk 1]
```diff
@@ -8,6 +8,7 @@ Changelog
 * :feature:`-`  The database size for users who have EVM transactions of any chain is now greatly reduced. Some databases had a reduction of up to 40%.
 * :feature:`6033` Optimism gas calculation is now correctly taking into account L1 gas fees.
 * :feature:`-` Optimism governance votes that include parameters should now be decoded properly.
+* :feature:`-` Delegations of the DIVA token are now properly shown in the decoded history events.
```
Comment (LefterisJP) [inline docs/changelog.rst:11]:
```suggestion
* :feature:`-` DIVA token airdrop claim and delegations are now properly shown in the decoded history events.
```

File: rotkehlchen/chain/ethereum/modules/diva/decoder.py [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+
+        event = self.base.make_event_from_transaction(
+            transaction=transaction,
+            tx_log=tx_log,
+            event_type=HistoryEventType.INFORMATIONAL,
+            event_subtype=HistoryEventSubType.GOVERNANCE,
+            asset=self.diva,
+            balance=Balance(),
+            location_label=event_address,
+            notes=f'{delegator} started to delegate on {self.diva.symbol} to {delegate}',
```
Comment (LefterisJP) [inline rotkehlchen/chain/ethereum/modules/diva/decoder.py:73]: Change the notes to use exactly same format as https://github.com/rotki/rotki/blob/8ff5b44bc4ff1940bf22cc6b80b2e486ceaf2a66/rotkehlchen/chain/optimism/modules/optimism/decoder.py#L47

We need to standardize all these. Both airdrop claims and delegations

Also don't make diva.symbol a variable. It's constant.

File: rotkehlchen/chain/ethereum/modules/diva/decoder.py [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+            counterparty=CPT_DIVA,
+        )
+        return DecodingOutput(event=event, refresh_balances=False)
+
+    def _decode_delegation_change(self, context: DecoderContext) -> DecodingOutput:
```
Comment (LefterisJP) [inline rotkehlchen/chain/ethereum/modules/diva/decoder.py:78]: Please put this in one function. Why did you make this extra function?

File: rotkehlchen/chain/ethereum/modules/diva/decoder.py [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+            if (
+                event.event_type == HistoryEventType.RECEIVE and
+                event.location_label == claiming_address and
+                event.asset == self.diva and
+                event.balance.amount == claimed_amount
+            ):
+                event.event_type = HistoryEventType.RECEIVE
+                event.event_subtype = HistoryEventSubType.AIRDROP
+                event.counterparty = CPT_DIVA
+                event.notes = f'Claim airdrop of {event.balance.amount} {self.diva.symbol}'
```
Comment (LefterisJP) [inline rotkehlchen/chain/ethereum/modules/diva/decoder.py:115]: Same as before for consistency. In optimism it's

`Claimed {amount} OP from optimism airdrop`

But perhas we should indeed change to present?

```suggestion
                event.notes = f'Claim {event.balance.amount} DIVA from the DIVA airdrop'
```

File: rotkehlchen/data/global.db
Comment (LefterisJP) [inline rotkehlchen/data/global.db:1]: What's the diff for the global DB?

**Review by LefterisJP — COMMENTED** [review]
small comments.

File: rotkehlchen/data/global.db
Comment (yabirgb) [reply rotkehlchen/data/global.db:1]: is in this message https://github.com/rotki/rotki/pull/6452#issuecomment-1665386368

File: rotkehlchen/chain/ethereum/modules/diva/decoder.py (hunk 3 above)
Comment (yabirgb) [reply rotkehlchen/chain/ethereum/modules/diva/decoder.py:78]: I forgot that we match against the address of the contract sending the log and not the transaction.to

**Review by LefterisJP — APPROVED** [review]
LGTM

**Pull request merged**